### PR TITLE
Added Coverage and Press-Release Filter

### LINKED ISSUE #2815: [News] Create to filter for news categories: "Press Releases" and "Coverage"
**Current Behavior**
The [layer5.io/news](https://layer5.io/news) page has no way of filtering between Layer5 produced news and news covered by external sources. The differences between these two news types is captured in the `category` metadata field.

**Desired Behavior**
- Add two filter buttons.
- Apply active state button coloring.
- Allow user to filter and unfilter on each category.

---
**Screenshots / Mockups**
Screen Shot 2022-05-19 at 5 15 04 PM

See https://www.figma.com/file/5ZwEkSJwUPitURD59YHMEN/Layer5-Designs?node-id=7978%3A13686

---

<h4>Contributor Resources</h4>

The layer5.io website uses Gatsby, React, and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- See [contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md)
- See Layer5 site designs in this [Figma project](https://www.figma.com/file/5ZwEkSJwUPitURD59YHMEN/Layer5-Designs). Fill-in the [Layer5 Community Member Form](https://layer5.io/newcomer) and join the [Layer5 Community Slack](http://slack.layer5.io) for access.

### PR DESCRIPTION
**Pull request by Harshit-Dandriyal**
Signed-off-by: Harshit Dandriyal <56961488+Harshit-Dandriyal@users.noreply.github.com>

**Description**
Added Search Input, coverage and press-release filter

Screenshot from 2022-05-21 04-09-56

This PR fixes #2815

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit a9c6d1877f0429f022f4b5e8f1d6e0fa4b61f364 at: https://62881bc85a97cd16d954ecc2--layer5ng.netlify.app

**Review by Akshit42-hue — CHANGES_REQUESTED** [review]
Hey @Harshit-Dandriyal great work, Just a small change when the tab is inactive change the color to grey, and when it is active change the color to black .

**Comment by Harshit-Dandriyal** [comment]
> Hey @Harshit-Dandriyal great work, Just a small change when the tab is inactive change the color to grey, and when it is active change the color to black .

Yes, both cards are active initially, that's why it's showing all cards :smile:

**Comment by Harshit-Dandriyal** [comment]
> > Hey @Harshit-Dandriyal great work, Just a small change when the tab is inactive change the color to grey, and when it is active change the color to black .
> 
> Yes, both cards are active initially, that's why it's showing all cards smile , when we click them they became inactive, color changes to gray and that card category will not be shown.

**Comment by l5io** [comment]
🚀 Preview for commit b113ae4cc47839c580bd88a89183f334591c20b3 at: https://6289e7a2837244268d7a7539--layer5ng.netlify.app

File: src/sections/Company/News-grid/NewsGrid.style.js
Comment (Chadha93) [inline src/sections/Company/News-grid/NewsGrid.style.js:29]: let's use an SVG here? referenced from assets?

File: src/sections/Company/News-grid/index.js
Comment (Chadha93) [inline src/sections/Company/News-grid/index.js:13]: Could make this a reusable function and import it wherever required to make this more robust? cc @Nikhil-Ladha your views?

**Review by Chadha93 — COMMENTED** [review]
Looks Good @Harshit-Dandriyal some suggestion. :)

File: src/sections/Company/News-grid/NewsGrid.style.js
Comment (Harshit-Dandriyal) [reply src/sections/Company/News-grid/NewsGrid.style.js:29]: Thanks, trying to implement it now. :smile:

File: src/sections/Company/News-grid/index.js
Comment (Harshit-Dandriyal) [reply src/sections/Company/News-grid/index.js:13]: I can make this reusable by implementing it in the custom button component that we made.

File: src/sections/Company/News-grid/index.js
Comment (Nikhil-Ladha) [inline src/sections/Company/News-grid/index.js:11]: Prefer `let` if at all you want to use a variable.

File: src/sections/Company/News-grid/index.js
Comment (Nikhil-Ladha) [inline src/sections/Company/News-grid/index.js:55]: camelCase please

File: src/sections/Company/News-grid/index.js
Comment (Nikhil-Ladha) [inline src/sections/Company/News-grid/index.js:72]: camelCase

File: src/sections/Company/News-grid/index.js
Comment (Nikhil-Ladha) [reply src/sections/Company/News-grid/index.js:13]: camelCase

File: src/sections/Company/News-grid/index.js
Comment (Nikhil-Ladha) [inline src/sections/Company/News-grid/index.js:45]: not needed

File: src/sections/Company/News-grid/index.js
Comment (Nikhil-Ladha) [inline src/sections/Company/News-grid/index.js:97]: We already have a search component. Please re-use that

**Review by Nikhil-Ladha — COMMENTED** [review]
We should be showing a text like "No more news to filter..." or something like that when both the filters are removed.

File: src/sections/Company/News-grid/index.js
Comment (Nikhil-Ladha) [inline src/sections/Company/News-grid/index.js:63]: Missing a `{}` here

File: src/sections/Company/News-grid/index.js
Comment (Nikhil-Ladha) [inline src/sections/Company/News-grid/index.js:48]: prefer camelCase

File: src/sections/Company/News-grid/index.js
Comment (Nikhil-Ladha) [inline src/sections/Company/News-grid/index.js:51]: Same as above

File: src/sections/Company/News-grid/index.js
Comment (Nikhil-Ladha) [inline src/sections/Company/News-grid/index.js:27]: Use camelCase while naming the state variables

**Comment by leecalcote** [comment]
> > Hey @Harshit-Dandriyal great work, Just a small change when the tab is inactive change the color to grey, and when it is active change the color to black .
> 
> Yes, both cards are active initially, that's why it's showing all cards 😄 , when we click the button they became inactive and that card category will not be shown.

Please display both cards as light gray initially. It shouldn’t be possible to have both buttons active concurrently.

**Comment by debo19** [comment]
@Harshit-Dandriyal lets discuss your updates on the websites call tomorrow

**Comment by Harshit-Dandriyal** [comment]
> @Harshit-Dandriyal lets discuss your updates on the websites call tomorrow

Actually, Tomorrow is my farewell in college still I will try my best to be present at the meeting. :smile:

**Comment by Harshit-Dandriyal** [comment]
Thanks, @leecalcote @Nikhil-Ladha @Chadha93  for a review, I will surely resolve all these issues by tomorrow.

**Comment by leecalcote** [comment]
@Harshit-Dandriyal feedback - https://www.youtube.com/watch?v=78Y9C0Qs58Y&t=1820s

**Comment by leecalcote** [comment]
@Harshit-Dandriyal thanks for today's demo in Meshery Dev Meeting. Reminder to follow up here.

**Comment by leecalcote** [comment]
A merge conflict has arisen now, @Harshit-Dandriyal

**Comment by leecalcote** [comment]
It's time for my daily check-in here. Not seeing any movement this weekend, @Harshit-Dandriyal, let's open this up for others to work on. // @layer5io/landscape-maintainers

**Comment by Harshit-Dandriyal** [comment]
Solving all issues now.

**Comment by l5io** [comment]
🚀 Preview for commit 1e30d1cc4dba8898bc0f2ae0f9b6d577d08d0352 at: https://6292a483d3a79e4eafa36291--layer5ng.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit aa50af6be9b883682ffadac9eeea6e0de6fd0f07 at: https://62931a22fe985411d8b838b4--layer5ng.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 1ca98dd468590c5d52d4a0dcbb3913f55befd5f5 at: https://6294b1630b69661ed95783a8--layer5ng.netlify.app

File: src/sections/Company/News-grid/index.js
Comment (Nikhil-Ladha) [inline src/sections/Company/News-grid/index.js:23]: Using classNames to achieve the same would be preferable, is what I think.
Like `ele.classList.add()` and `ele.classList.remove()`

Comment (Harshit-Dandriyal) [reply src/sections/Company/News-grid/index.js:23]: working on this :+1:

**Comment by Harshit-Dandriyal** [comment]
@leecalcote @Nikhil-Ladha Here is the idea I was talking about, This is just a prototype of filtering data on the basis of the content, Hence it is a prototype we are only filtering for the first two lines of our content of news, Is it working fine as we want?

**Comment by l5io** [comment]
🚀 Preview for commit 48d20e032f787c9ba2dbb79fb5137f5d6b975a1a at: https://629519471f6a480dda096398--layer5ng.netlify.app

**Comment by kayceeDev** [comment]
While this solution might work, don't you think it will be tedious to add the body field to all `index.mdx` Won't it be better to add the `body` in the graphQL query? It will return a bunch of string that also contains all the contents of the body. This way, we can also search on the body. @Harshit-Dandriyal

**Comment by Nikhil-Ladha** [comment]
I second @kayceeDev here.
This is a tedious approach and will require updating all the files with this extra front matter. But, the suggested approach only requires you to add another field `body` in the query and use in it the dataList that the plugin receives. That will index the body of the blog/news/resource and allow us to search through it as well.

**Comment by kayceeDev** [comment]
> I second @kayceeDev here. This is a tedious approach and will require updating all the files with this extra front matter. But, the suggested approach only requires you to add another field `body` in the query and use in it the dataList that the plugin receives. That will index the body of the blog/news/resource and allow us to search through it as well.

Spot on @Nikhil-Ladha . I already tried this locally and it seems to work as expected. I'd look further and then make a PR for the blog page.

**Comment by l5io** [comment]
🚀 Preview for commit 46815e37004a89032d09d92beb77e565e7d223d1 at: https://6298aaa18c0997114baa3649--layer5ng.netlify.app

**Review by Chadha93 — COMMENTED** [review]
@Harshit-Dandriyal pr ready? Let's get you covered on DCO :)

**Comment by kayceeDev** [comment]
> @Harshit-Dandriyal pr ready? Let's get you covered on DCO :)

I don't think it is. One, the search feature doesn't return the required results. Secondly, it's not mobile responsive
Screenshot_20220602-142450_Chrome

**Comment by l5io** [comment]
🚀 Preview for commit de870d42c945c05cdbd4fba75c8ad387645ca71e at: https://629924bdaaba01046b80a514--layer5ng.netlify.app

**Comment by kayceeDev** [comment]
Good job. Seems like you still need to adjust responsiveness.

**Comment by Harshit-Dandriyal** [comment]
dragon-ball-super-saiyan

**Comment by l5io** [comment]
🚀 Preview for commit 2d6c58402068f56fcca5c76f72589a2482d0c948 at: https://62992ae203a4c70c29552a16--layer5ng.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 2d6c58402068f56fcca5c76f72589a2482d0c948 at: https://629a60d6ead3d20dd601fc8c--layer5.netlify.app

**Review by Chadha93 — APPROVED** [review]
@Harshit-Dandriyal Good work :100: LGTM, let's get all approval and merge this feature. Yay!!

**Comment by Harshit-Dandriyal** [comment]
> @Harshit-Dandriyal Good work 100 LGTM, let's get all approval and merge this feature. Yay!!

Thanks, Finally :blush: :relaxed:

**Review by kayceeDev — APPROVED** [review]
Well done @Harshit-Dandriyal 👏

**Pull request merged**
